proto/openconfig/enums/gen.go

```diff
@@ -0,0 +1,21 @@
+//
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package enums contains code generated from the Hercules OpenConfig enums
+// protobuf.
+package enums
+
+// pb.go can be generated with go generate.
+//go:generate sh -c "cd $GOPATH/src && protoc --proto_path=. --go_out=:. github.com/openconfig/hercules/proto/openconfig/enums/enums.proto"
```

Comment:
What is the role of these gen.go files? They all (in various sub-directories) seems to have the same name. And it is unclear why these lines are commented out in each.
(Sorry, I'm not a go person)

Reply:
This commented out line allows one to just type "go generate" in each directory and it generates the .pb.go files.